### PR TITLE
[EMCAL-529] Change NumPatchesPerFastORCheck to nsigma over mean checker

### PR DESCRIPTION
**Pull request by sweyh99**
Determine if a FastOR is noisy by how many sigma it is over the mean FastOR noise level 
Remove from the list of noisy FastORs any neighboring FastORs to the noisiest ones

### CODE REVIEW

**Comment by knopers8** [comment]
@mfasDa this seems waiting for your review

File: Modules/EMCAL/src/NumPatchesPerFastORCheck.cxx [hunk 1]
```diff
@@ -117,43 +112,121 @@ Quality NumPatchesPerFastORCheck::check(std::map<std::string, std::shared_ptr<Mo
...
-                if (mLogLevelIL > 2) {
-                  ILOG(Warning, Support) << errorMessage << ENDM;
-                }
-                mHighCountTRUPositions.insert(obj);
+            }
+          }
+
+          // Save the final bad FastORs
+          FastORNoiseLevel final{ i->mCounts, i->mFastORID, i->mPosGlobalPhi, i->mPosGlobalEta, i->mRejected };
+          finalBadFastORs.push_back(final);
```
Comment (mfasDa) [inline Modules/EMCAL/src/NumPatchesPerFastORCheck.cxx:173]: `final` is a reserved keyword, please replace.

**Review by mfasDa — CHANGES_REQUESTED** [review]
One small issue, otherwise it is in line with what we discussed and could be merged.

**Review by mfasDa — APPROVED** [review]
PR is approved, the overwriting of the keyword should be done in a separate commit.

**Pull request merged**